apps/src/signUpFlow/FinishStudentAccount.tsx

```diff
@@ -45,6 +47,44 @@ const FinishStudentAccount: React.FunctionComponent<{
   const [showAgeError, setShowAgeError] = useState(false);
   const [showStateError, setShowStateError] = useState(false);
 
+  const [gdprChecked, setGdprChecked] = useState(false);
+  const [showGDPR, setShowGDPR] = useState(false);
+  const [isGdprLoaded, setIsGdprLoaded] = useState(false);
+
+  useEffect(() => {
+    const fetchGdprData = async () => {
+      const urlParams = new URLSearchParams(window.location.search);
+      const forceInEu = urlParams.get('force_in_eu');
+      try {
+        const response = await fetch(
+          `/users/gdpr_check?force_in_eu=${forceInEu}`
+        );
+        const data = await response.json();
+        if (data.gdpr || data.force_in_eu === '1') {
+          setShowGDPR(true);
+        }
+      } catch (error) {
+        console.error('Error fetching GDPR data:', error);
+      } finally {
+        setIsGdprLoaded(true);
+      }
+    };
+    fetchGdprData();
+  }, []);
+
+  // GDPR is valid if
+  // 1. The fetch call has completed AND
+  //   2. GDPR is showing AND checked OR
+  //   3. GDPR is not relevant (not showing)
+  const gdprValid = useMemo(() => {
```

Comment:
@drizco I just wanted to give you a shoutout here- this is the first time I've used useMemo() after hearing you talk about it!